src/Cardano/Wallet/Api/Types.hs

```diff
@@ -38,15 +38,9 @@ module Cardano.Wallet.Api.Types
     , WalletPutPassphraseData (..)
 
     -- * Encoding & Decoding
-    , DecodeApiAddressError (..)
-    , decodeApiAddress
-    , encodeApiAddress
-    , DecodeApiEncryptionPassphraseError (..)
-    , decodeApiEncryptionPassphrase
-    , encodeApiEncryptionPassphrase
-    , DecodeApiWalletNameError (..)
-    , decodeApiWalletName
-    , encodeApiWalletName
+    , ToText (..)
+    , FromText (..)
+    , TextDecodingError (..)
```

Comment:
The overall looks clean and re-usable. Nice work! 
(one reserve about the `TextDecodingError`)

Reply:
Thanks! I've moved the textual encoding work into a separate PR: #142.

I'll keep this PR for the `docopt` CLI-specific work.